src/backends/torch/torchinputconns.cc

```diff
@@ -306,18 +308,26 @@ namespace dd
                   }
 
                 // Read data
-                for (const std::pair<std::string, std::vector<double>> &lfile :
-                     lfiles)
+                if (_db)
                   {
-                    add_image_file<std::vector<double>>(_dataset, lfile.first,
-                                                        lfile.second);
+                    for (const std::pair<std::string, std::vector<double>>
+                             &lfile : lfiles)
+                      {
+                        _dataset.add_image_file(lfile.first, lfile.second,
```

Comment:
linked to same comment : here you pass `_height` and `_width` from this (inputc) , but adding image needs other params from `inputc` that will be read through  `_dataset._inputc`